src/AzureCacheOptions.cs

```diff
@@ -38,10 +38,15 @@ public class AzureCacheOptions
     public bool ThrowOnTokenRefreshFailure = true;
 
     /// <summary>
-    /// How long before expiration should a token be refreshed. 
-    /// Tokens have a 24hr lifespan by default, and the default margin of 4 hours allows time for recovery of any issues preventing refresh before the token expires.
+    /// Determines whether the current token should be refreshed, based on its age and lifespan
     /// </summary>
-    internal TimeSpan TokenExpirationMargin = TimeSpan.FromHours(4);
+    public Func<DateTime, DateTime, bool> ShouldTokenBeRefreshed = (DateTime acquired, DateTime expiry) =>
```

Comment:
Just double checking, this shouldn't cause a token refresh to happen more than every 5 minutes since we have the TokenHeartbeatInterval set to 5 mins correct?

Reply:
True, the 5m heartbeat interval dictates the minimum interval for token refresh

Reply:
I'm not sure if it would be practical, but would there be cases where an identity token could expire in that short time? I was just wondering what would happen if a clients identity caused a lot of requests, but I might not have enough context on how the token creation works.

Reply:
Minimum token lifetime is 10m so we should be okay: https://learn.microsoft.com/en-us/azure/active-directory/develop/configurable-token-lifetimes#access-id-and-saml2-token-lifetime-policy-properties

Reply:
Thats a good idea. We can gather data once AAD is fully rolled out on thresholds so we can configure an advisor and/or a monitor if needed.